src/tools/compiletest/src/main.rs

```diff
@@ -465,6 +465,8 @@ pub fn test_opts(config: &Config) -> test::TestOpts {
             Err(_) => false,
         },
         color: config.color,
+        #[cfg(not(bootstrap))]
+        pretty_print_assertions: false,
```

Comment:
This looks odd - is compiletest _ever_ built with cfg(bootstrap)? It's a stage 0 tool, I wouldn't expect it to be compiled twice.

Reply:
I don't really know to be honest, but the CI failures seemed to indicate that compiletest was being compiled with both the old and new `test` crate at different points, in both cases with `cfg(bootstrap)`.

I don't see a way to find the old CI results now, or I would link to them, but it really didn't pass without this.

Reply:
I think the fix for this is to remove `cfg(not(bootstrap))` from the standard library, not to add it in more places. I would be very surprised if compiletest was linking to two different versions of libtest.

Reply:
That's how I started, without any `#[cfg(not(bootstrap))]`. I only added it to get past CI: https://github.com/rust-lang/rust/pull/79001/commits/522710ea1babcdad2553434332fa577c72f813c0

Reply:
Oh, and to be clear, then I first added it only to `compiletest`, which got it past some tests, but then a later test failed until I also added it to `libtest`.

Reply:
Hmm, ok. I can't see the CI logs for that but if it broke without cfgs I guess this is ok to add, it will only be around for 6 weeks anyway.

Reply:
Looks like https://github.com/rust-lang/rust/pull/68019 should have fixed this. But if this fails without `cfg` then maybe it didn't? @cuviper, is it possible that PR didn't fix the problem entirely?

Reply:
> The second failure only triggered with --all-targets.

This sounds like a cross-compilation issue - maybe compile test is being linked to stage 1 when running tests on stage2? That seems like the wrong behavior, since it always runs on the host.

Reply:
Could certainly be. I'm not familiar enough with the build system to give useful insight though :(

Reply:
@m-ou-se I think #68019 is still doing the right thing, because this isn't dealing with stage0's own `std`/`test`.

If the `test` crate _built_ by stage0 has some functionality gated by `cfg(bootstrap)`, then `compiletest` will have to match. However, it's not clear to me why `test` would need to be gated this way at all. Is it using some language feature that's only supported in the newest compiler? What were the errors that prompted you to add `bootstrap` configs at all?

Reply:
The error was that `TestOpts` didn't have the `pretty_print_assertions` field, so it's not about compiler features.

I managed to dig up an URL to the test result from my mailbox: https://github.com/rust-lang/rust/runs/1392779322?check_suite_focus=true#step:24:1853

Then, after adding `#[cfg(not(bootstrap))]` only in `compiletest`, this one: https://github.com/rust-lang/rust/runs/1393384004?check_suite_focus=true#step:24:1368

Reply:
OK, I'll try to see if I can figure out what's going on without those cfgs...

Reply:
I think I see it -- the failure came from `python3 ../x.py test --stage 0 src/tools/compiletest`, which is using the original stage0 sysroot instead of the newly built `stage0-sysroot`. You can see this with extra verbose `-v -v` flags. So yes, this does appear to be something that #68019 missed -- I'll see about fixing that.

Reply:
Cool, thanks for investigating. If the fix happens to land before this, I'll remove all the cfgs :)

Reply:
See #79183.

Reply:
https://github.com/rust-lang/rust/pull/79183 merged, so I think you can remove the cfgs now :)

Reply:
Indeed, I'll get rid of those too when resolving the conflicts :)